csharp/src/Apache.Arrow/Arrays/Array.cs

```diff
@@ -60,6 +60,20 @@ internal static void Accept<T>(T array, IArrowArrayVisitor visitor)
             }
         }
 
+        public Array Slice(int offset, int length)
+        {
+            if (offset > Length)
+            {
+                throw new ArgumentException($"Offset {offset} cannot be greater than Length {Length} for Array.Slice");
+            }
+
+            length = Math.Min(Data.Length - offset, length);
+            offset += Data.Offset;
+
+            ArrayData newData = Data.Slice(offset, length);
+            return ArrowArrayFactory.BuildArray(newData) as Array;
```

Comment:
IMO- We should really get rid of IArrowArray, and just use the base `Array` class. I don't think having both does any good. I'll open a JIRA for this.